Libraries/LibDNS/Resolver.h

```diff
@@ -476,11 +611,471 @@ class Resolver {
+                        dbgln_if(DNS_DEBUG, "DNS: No DNSKEY found for validation of {} RRs", rrset_with_rrsig.rrset.size());
+                        continue;
+                    }
+
+                    promises.append(validate_rrset_with_rrsig(move(rrset_with_rrsig), result));
+                }
+
+                auto promise = Core::Promise<Empty>::after(move(promises))
+                                   ->when_resolved([result, lookup, keys = move(keys)](Empty) {
+                                       for (auto& key : keys)
+                                           result->add_dnskey(key);
+                                       result->set_dnssec_validated(true);
+                                       result->set_being_dnssec_validated(false);
+                                       result->finished_request();
+                                       lookup.promise->resolve(result);
+                                   })
+                                   .when_rejected([result, lookup](Error& error) {
+                                       result->finished_request();
+                                       result->set_being_dnssec_validated(false);
+                                       lookup.promise->reject(move(error));
+                                   })
+                                   .map<NonnullRefPtr<LookupResult const>>([result](Empty&) { return result; });
+
+                lookup.promise = move(promise);
+            };
+
+            if (is_root_zone) {
+                resolve_using_keys(Vector<Messages::Records::DNSKEY> {
+                    {
+                        .flags = 257,
+                        .protocol = 3,
+                        .algorithm = Messages::DNSSEC::Algorithm::RSASHA256,
+                        .public_key = MUST(decode_base64("AwEAAaz/tAm8yTn4Mfeh5eyI96WSVexTBAvkMgJzkKTOiW1vkIbzxeF3+/4RgWOq7HrxRixHlFlExOLAJr5emLvN7SWXgnLh4+B5xQlNVz8Og8kvArMtNROxVQuCaSnIDdD5LKyWbRd2n9WGe2R8PzgCmr3EgVLrjyBxWezF0jLHwVN8efS3rCj/EWgvIWgb9tarpVUDK/b58Da+sqqls3eNbuv7pr+eoZG+SrDK6nWeL3c6H5Apxz7LjVc1uTIdsIXxuOLYA4/ilBmSVIzuDWfdRUfhHdY6+cn8HFRm+2hM8AnXGXws9555KrUB5qihylGa8subX2Nn6UwNR1AkUTV74bU="sv)),
+                        .calculated_key_tag = 20326,
+                    },
+                    {
+                        .flags = 256,
+                        .protocol = 3,
+                        .algorithm = Messages::DNSSEC::Algorithm::RSASHA256,
+                        .public_key = MUST(decode_base64("AwEAAa96jeuknZlaeSrvyAJj6ZHv28hhOKkx3rLGXVaC6rXTsDc449/cidltpkyGwCJNnOAlFNKF2jBosZBU5eeHspaQWOmOElZsjICMQMC3aeHbGiShvZsx4wMYSjH8e7Vrhbu6irwCzVBApESjbUdpWWmEnhathWu1jo+siFUiRAAxm9qyJNg/wOZqqzL/dL/q8PkcRU5oUKEpUge71M3ej2/7CPqpdVwuMoTvoB+ZOT4YeGyxMvHmbrxlFzGOHOijtzN+u1TQNatX2XBuzZNQ1K+s2CXkPIZo7s6JgZyvaBevYtxPvYLw4z9mR7K2vaF18UYH9Z9GNUUeayffKC73PYc="sv)),
+                        .calculated_key_tag = 38696,
+                    },
+                });
```

Comment:
We should probably be getting the root keys straight from IANA, not sure if this should be done at build time, or at runtime 🤔